readthedocs_ext/readthedocs.py

```diff
@@ -172,12 +182,61 @@ def generate_json_artifacts(app, pagename, templatename, context, doctree):
         log.exception(
             'Fail to save JSON output for page {page}'.format(page=outjson)
         )
-    except Exception as e:
+    except Exception:
         log.exception(
             'Failure in JSON search dump for page {page}'.format(page=outjson)
         )
 
 
+def dump_sphinx_data(app, exception):
+    """
+    Dump a bunch of additional Sphinx data that is useful during search indexing
+    """
+    if app.builder.name not in JSON_BUILDERS:
+        return
+    try:
+        types = {}
+        for domain_name, domain_obj in app.env.domains.items():
+            for type_name, type_obj in domain_obj.object_types.items():
+                key = "{}:{}".format(domain_name, type_name)
+                types[key] = str(type_obj.lname)
+
+        titles = {}
+        for page, title in app.env.titles.items():
+            titles[app.builder.get_target_uri(page)] = title.astext()
+
+        paths = {}
+        for page, title in app.env.titles.items():
+            paths[app.builder.get_target_uri(page)] = app.env.doc2path(page, base=None)
```

Comment:
We can merge this with the other for loop, but I guess this is more readable, and not so much difference in the performance with our big servers  p:

Reply:
I think it's better to merge the loops as well. We can also calculate the `get_target_uri` once only.